specs/simpleserialize.md

```diff
@@ -0,0 +1,302 @@
+|:---------------|:-----:|:------------------------------------------------------------------------|
+| `LENGTH_BYTES` |   4   | Number of bytes used for the length added before the serialized object. |
+
+
+## Overview
+
+### Serialize/Encode
+
+#### uint: 8/16/24/32/64/256
+
+Convert directly to bytes the size of the int. (e.g. ``uint16 = 2 bytes``)
+
+All integers are serialized as **big endian**.
+
+| Check to perform                 | Code                    |
+|:---------------------------------|:------------------------|
+| Size is a byte integer           | ``int_size % 8 == 0``   |
+| Value is less than max           | ``2**int_size > value`` |
+
+```python
+buffer_size = int_size / 8
+return value.to_bytes(buffer_size, 'big')
+```
+
+#### Address
+
+The address should already come as a hash/byte format. Ensure that length is
+**20**.
+
+| Check to perform       | Code                 |
+|:-----------------------|:---------------------|
+| Length is correct (20) | ``len(value) == 20`` |
+
+```python
+assert( len(value) == 20 )
+return value
+```
+
+#### Hash
+
+| Hash Type | Usage                                           |
+|:---------:|:------------------------------------------------|
+|  `hash32` | Hash size of ``keccak`` or `blake2b[0.. < 32]`. |
+|  `hash96` | BLS Public Key Size.                            |
+|  `hash97` | BLS Public Key Size with recovery bit.          |
+
+
+| Checks to perform                   | Code                 |
+|:-----------------------------------|:---------------------|
+| Length is correct (32) if `hash32` | ``len(value) == 32`` |
+| Length is correct (96) if `hash96` | ``len(value) == 96`` |
+| Length is correct (97) if `hash97` | ``len(value) == 97`` |
+
+
+**Example all together**
+
+```python
+if (type(value) == 'hash32'):
+   assert(len(value) == 32)
+elif (type(value) == 'hash96'):
+   assert(len(value) == 96)
+elif (type(value) == 'hash97'):
+   assert(len(value) == 97)
+else:
+   raise TypeError('Invalid hash type supplied')
+
+return value
+```
+
+##### Hash32
+
+Ensure 32 byte length and return the bytes.
+
+```python
+assert(len(value) == 32)
+return value
+```
+
+##### Hash96
+
+Ensure 96 byte length and return the bytes.
+
+```python
+assert(len(value) == 96)
+return value
+```
+
+##### Hash97
+
+Ensure 97 byte length and return the bytes.
+
+```python
+assert(len(value) == 97)
+return value
+```
+
+#### Bytes
+
+For general `byte` type:
+1. Get the length/number of bytes; Encode into a 4 byte integer.
+2. Append the value to the length and return: ``[ length_bytes ] + [
+   value_bytes ]``
+
+| Check to perform                     | Code                   |
+|:-------------------------------------|:-----------------------|
+| Length of bytes can fit into 4 bytes | ``len(value) < 2**32`` |
+
+```python
+byte_length = (len(value)).to_bytes(LENGTH_BYTES, 'big')
+return byte_length + value
+```
+
+#### List/Vectors
+
```

Comment:
Should add "Length of list can fit into 4 bytes" as a check in a "Check to perform" table as well as an assertion in the sample code